### PR TITLE
Fix wallet P3A reporting on Android, fix wallet P3A "no report" period

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#21448

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See issue.

### CODE REVIEW

File: browser/profiles/brave_profile_manager.cc [hunk 1]
```diff
@@ -120,6 +121,7 @@ void BraveProfileManager::DoFinalInitForServices(Profile* profile,
     return;
   brave_ads::AdsServiceFactory::GetForProfile(profile);
   brave_rewards::RewardsServiceFactory::GetForProfile(profile);
+  brave_wallet::BraveWalletServiceFactory::GetServiceForContext(profile);
```
Comment (DJAndries) [inline browser/profiles/brave_profile_manager.cc:124]: Will start wallet service at the start of the browser session to keep P3A values up to date

File: browser/brave_wallet/brave_wallet_service_unittest.cc [hunk 2]
```diff
@@ -1226,7 +1226,7 @@ TEST_F(BraveWalletServiceUnitTest, RecordWalletNoUse) {
   histogram_tester_->ExpectTotalCount(kBraveWalletWeeklyHistogramName, 0);
   histogram_tester_->ExpectTotalCount(kBraveWalletMonthlyHistogramName, 0);
 
-  task_environment_.FastForwardBy(base::Days(5));
+  task_environment_.FastForwardBy(base::Days(4));
```
Comment (rillian) [inline browser/brave_wallet/brave_wallet_service_unittest.cc:1229]: What does this change do?

File: browser/profiles/brave_profile_manager.cc (hunk 1 above)
Comment (rillian) [reply browser/profiles/brave_profile_manager.cc:124]: Thanks for the comment. It's good to put something like this in the body of the commit message, so the motivation of a change is easier to track down when reading the code later.

**Review by rillian — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

File: browser/brave_wallet/brave_wallet_service_unittest.cc (hunk 2 above)
Comment (DJAndries) [reply browser/brave_wallet/brave_wallet_service_unittest.cc:1229]: We have a one week "no report" period. In this test, I advance the time by three days, and then by five days, which is eight days total. The "no report" period should end after seven days.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by GeetaSarvadnya** [comment]
Verification PASSED on `Samsung Tab A10 version 11` running `Bravemonoarmx64.apk_1.38.34`

- Verified the failed cases 1 and 9 from https://github.com/brave/brave-browser/issues/21063#issuecomment-1058097153 and ensured that the cases are working as expected.

<details>
<summary>Case 1: Brave.Wallet.UsageDaysInWeek=0 _PASSED</summary>

- Set device date to March 1st
- Clean install and enable walelt
- Move device date ahead by 1 week
- Relaunch browser and check local-state, doesn't list `Brave.Wallet.UsageDaysInWeek`
- Quit browser and move device date ahead by 1 day
- Relaunch browser and check local-state, shows `Brave.Wallet.UsageDaysInWeek=1`
- Quit browser and move device date ahead to next Monday
- Relaunch browser and check local-state, the metrics shows `Brave.Wallet.UsageDaysInWeek=0`

8th March | 16th March
-----|------
Screenshot_20220308-135839_Brave - Nightly | Screenshot_20220316-140004_Brave - Nightly



</details>

<details>
<summary>Case 9: </summary>

</details>

**Pull request merged**